plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -0,0 +1,180 @@
+   *
+   * @param settings settings.
+   */
+  public void loadConnectors(Settings settings) {
+    doPrivileged(() -> {
+      Boolean isPPLEnabled = (Boolean) OpenSearchSettings.PPL_ENABLED_SETTING.get(settings);
+      Boolean isFederationEnabled = CatalogSettings.FEDERATION_ENABLED.get(settings);
+      if (isPPLEnabled && isFederationEnabled) {
+        InputStream inputStream = CatalogSettings.CATALOG_CONFIG.get(settings);
+        if (inputStream != null) {
+          ObjectMapper objectMapper = new ObjectMapper();
+          objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+          try {
+            ArrayNode catalogs = objectMapper.readValue(inputStream, ArrayNode.class);
+            validateCatalogs(catalogs);
+            constructConnectors(catalogs);
+          } catch (IOException e) {
+            LOG.error("Catalog Configuration File uploaded is malformed. Verify and re-upload.");
+            throw new IllegalArgumentException(
+                "Malformed Catalog Configuration Json" + e.getMessage());
+          }
+        }
+      } else {
+        LOG.info("PPL federation is not enabled.");
+      }
+      return null;
+    });
+  }
+
+  @Override
+  public Optional<StorageEngine> getStorageEngine(String catalog) {
+    return Optional.ofNullable(storageEngineMap.get(catalog));
+  }
+
+  @Override
+  public Set<String> getCatalogs() {
+    return storageEngineMap.keySet();
+  }
+
+  private <T> T doPrivileged(PrivilegedExceptionAction<T> action) {
+    try {
+      return SecurityAccess.doPrivileged(action);
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to perform privileged action", e);
+    }
+  }
+
+  private StorageEngine createStorageEngine(JsonNode catalog) throws URISyntaxException {
+    StorageEngine storageEngine;
+    String connector = catalog.get(CatalogConstants.CONNECTOR).asText();
+    switch (connector) {
+      case "prometheus":
+        storageEngine = null;
+        break;
+      default:
+        LOG.info(
+            "Unknown connector \"{}\". "
+                + "Please re-upload catalog configuration with a supported connector.",
+            catalog.get("connector"));
+        throw new IllegalStateException(
+            "Unknown connector. Connector doesn't exist in the list of supported.");
+    }
+    return storageEngine;
+  }
+
+  private void constructConnectors(ArrayNode catalogs) throws URISyntaxException {
+    storageEngineMap = new HashMap<>();
+    for (JsonNode catalog : catalogs) {
+      String catalogName = catalog.get(CatalogConstants.NAME).asText();
+      StorageEngine storageEngine = createStorageEngine(catalog);
+      storageEngineMap.put(catalogName, storageEngine);
+    }
+  }
+
+  /**
+   * This can be moved to a different validator class
+   * when we introduce more connectors.
+   *
+   * @param catalogs catalogs.
+   */
+  private void validateCatalogs(ArrayNode catalogs) {
```

Comment:
add UT?

Reply:
Ok...somehow missed. Will add.

Reply:
is it possible to bind the json to an object and use Jackson's annotation to do the validation? https://stackoverflow.com/questions/36357713/java-8-jackson-validation